components/transaction/internal/adapters/rabbitmq/healthcheck.go

```
@@ -0,0 +1,213 @@
+// It starts/stops the health checker based on circuit state transitions.
+//
+// State transitions:
+//   - Any circuit opens (CLOSED → OPEN): Start health checker
+//   - Circuit goes half-open (OPEN → HALF-OPEN): Keep running
+//   - All circuits close (→ CLOSED): Stop health checker
+func (s *StateAwareHealthChecker) OnStateChange(serviceName string, from, to libCircuitBreaker.State) {
+	// Log state change outside the critical section to reduce lock contention
+	s.logger.Debugf("StateAwareHealthChecker: %s state changed %s -> %s", serviceName, from, to)
+
+	var shouldStart, shouldStop bool
+	var logServiceName string
```

golangci-lint (2.5.0)

[error] missing whitespace above this line (never cuddle decl)

(wsl_v5)